src/runaqa.ts

```diff
@@ -1,324 +1,475 @@
+      await exec.exec('make', [`${target}`, `${customOption}`], options)
+    }
+    else if (target.includes('-f parallelList.mk')){
+      // move the parallelList to TKG/
+      if (IS_WINDOWS){
+        await io.cp(
+          `${process.env.GITHUB_WORKSPACE}\\parallelList.mk`,
+          `${process.env.GITHUB_WORKSPACE}\\aqa-tests\\TKG\\parallelList.mk`
+        )
+      } else {
+        await io.cp(
+          `${process.env.GITHUB_WORKSPACE}/parallelList.mk`,
+          `${process.env.GITHUB_WORKSPACE}/aqa-tests/TKG/parallelList.mk`
+        )
+      }
+      await exec.exec(`make ${target}`);
+   } else {
+      await exec.exec('make', [`${target}`], options)
+    }
+  } catch (error) {
+    if (error instanceof Error){
+      core.setFailed(error.message)
+    }
+    else{
+      core.setFailed('Unexpected error')
+    }
+  }
+  if (myOutput.includes('FAILED test targets') === true) {
+    core.setFailed('There are failed tests')
+  }
+}
+
+/**
+ * Read job.properties and reset TEST_JDK_HOME env variable.
+ * @return {[null]}  null
+ */
+function resetJDKHomeFromProperties(): void {
+  const jobProperties = `${process.env.GITHUB_WORKSPACE}/aqa-tests/job.properties`
+  if (fs.existsSync(jobProperties)) {
+    const lines = fs
+      .readFileSync(jobProperties, 'utf-8')
+      .replace(/\r\n/g, '\n')
+      .split('\n')
+      .filter(Boolean)
+    for (const l of lines) {
+      const regexp = /TEST_JDK_HOME=(.*)/
+      const match = regexp.exec(l)
+      if (match && match[1]) {
+        process.env.TEST_JDK_HOME = match[1]
+        core.info(`Reset TEST_JDK_HOME to ${process.env.TEST_JDK_HOME}`)
+      }
+    }
+  }
+}
+
+/**
+ * Sets javaHome on the runner. Raise exception if not able to be set.
+ * @param  {string} version JDK version
+ * @param  {string} jdksource [description]
+ * @return {[string]}  javaHome    Java home
+ */
+function getTestJdkHome(version: string, jdksource: string): string {
+  // Try JAVA_HOME first and then fall back to GITHUB actions default location
+  let javaHome = process.env[`JAVA_HOME_${version}_X64`] as string
+  if (javaHome === undefined) {
+    javaHome = process.env['JAVA_HOME'] as string
+  }
+  if (jdksource === 'install-jdk') {
+    // work with AdoptOpenJDK/install-sdk
+    if (`JDK_${version}` in process.env) {
+      javaHome = process.env[`JDK_${version}`] as string
+    } else {
+      javaHome = process.env.JAVA_HOME as string
+    }
+  }
+  // Remove spaces in Windows path and replace with a short name path, e.g. 'C:/Program Files/***' ->C:/Progra~1/***
+  if (IS_WINDOWS && jdksource === 'github-hosted') {
+    javaHome = javaHome.replace(/Program Files/g, 'Progra~1')
+  }
+  if (javaHome === undefined) {
+    // exit with error if JDK cannot be found
+    core.setFailed('JDK could not be found')
+  }
+  return javaHome
+}
+
+/**
+ * This function is an alternative of extra install step in workflow or alternative install action. This could also be implemented as github action

```

Comment:
I's add an issue in the run-aqa repo for this (create extraGH Action)

Reply:
Will do